core-bundle/src/EventListener/CsrfTokenCookieListener.php

```diff
@@ -57,8 +60,38 @@ public function onKernelResponse(FilterResponseEvent $event): void
             return;
         }
 
-        $request = $event->getRequest();
-        $response = $event->getResponse();
+        if ($this->requiresCsrf($event->getRequest(), $event->getResponse())) {
+            $this->setCookies($event->getRequest(), $event->getResponse());
+        } else {
+            $this->removeCookies($event->getRequest(), $event->getResponse());
+        }
+    }
+
+    private function requiresCsrf(Request $request, Response $response): bool
+    {
+        foreach ($request->cookies as $key => $value) {
+            if (!$this->isCsrfCookie($key, $value)) {
+                return true;
+            }
+        }
+
+        if (\count($response->headers->getCookies(ResponseHeaderBag::COOKIES_ARRAY))) {
+            return true;
+        }
+
+        if ($request->getUserInfo()) {
```

Comment:
This is correct. CSRF protection is needed for any case where the browser can cause automatic authentication which is usually either a cookie or stored basic auth.